src/engine.cpp

```diff
@@ -16,53 +16,77 @@ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLI
 #include <fstream>
 #include <functional>
 #include <random>
+#include <filesystem>
+
 //Create definitions of the RandomEngine class here
 //Methods:
 RandomEngine::RandomEngine():length((uint8_t)8),symb(false),mod(symb?73:63){
     //Reading 100 bytes of data from urandom file for seed generation.
     //this part is independent of user's input, hence defined in the default contructor.
     std::ifstream file("/dev/urandom", std::ios::in | std::ios::binary);
     if(!file){
-    std::cout << "Error opening file\n";
+    std::cout<<"Error opening file\n";
     }
     file.read(seq, sizeof(char)*100);
     file.close();
     std::seed_seq seed(seq, seq+100);
     gen= new std::mt19937_64(seed);
     dist = new std::uniform_int_distribution<int64_t>(0,mod);
+
+    if(!gen || !dist){
+            throw std::runtime_error("Cannot generate password\n");
+    }
 }
 RandomEngine::RandomEngine(uint8_t l,bool s): length(l), symb(s),mod(symb?73:63){
     std::ifstream file("/dev/urandom", std::ios::in | std::ios::binary);
     if(!file){
-    std::cout << "Error opening file\n";
+    std::cout<<"Error opening file\n";
     }
     file.read(seq, sizeof(char)*100);
     file.close();
     std::seed_seq seed(seq, seq+100);
     gen= new std::mt19937_64(seed);
     dist = new std::uniform_int_distribution<int64_t>(0,mod);
+
+    if(!gen || !dist){
+            throw std::runtime_error("Cannot generate password\n");
+    }
 }
 RandomEngine::RandomEngine(uint8_t l):length(l),mod(symb?73:63){
     std::ifstream file("/dev/urandom", std::ios::in | std::ios::binary);
     if(!file){
-    std::cout << "Error opening file\n";
+    std::cout<<"Error opening file\n";
     }
     file.read(seq, sizeof(char)*100);
     file.close();
     std::seed_seq seed(seq, seq+100);
     gen= new std::mt19937_64(seed);
     dist = new std::uniform_int_distribution<int64_t>(0,mod);
+
+    if(!gen || !dist){
+            throw std::runtime_error("Cannot generate password\n");
+    }
 }
-  RandomEngine::RandomEngine(bool s): symb(s),mod(symb?73:63){
+RandomEngine::RandomEngine(bool s): symb(s),mod(symb?73:63){
      std::ifstream file("/dev/urandom", std::ios::in | std::ios::binary);
     if(!file){
-    std::cout << "Error opening file\n";
+    std::cout<<"Error opening file\n";
     }
     file.read(seq, sizeof(char)*100);
     file.close();
     std::seed_seq seed(seq, seq+100);
     gen= new std::mt19937_64(seed);
     dist = new std::uniform_int_distribution<int64_t>(0,mod);
+
+    if(!gen || !dist){
+            throw std::runtime_error("Cannot generate password\n");
+    }
+}
+
+RandomEngine::~RandomEngine(){
```

Comment:
Maybe define the destructor after the ```getstring()``` function.

Reply:
Yes, because we have to make sure that ```dist``` and ```gen``` are NOT freed before the ```getstring()``` function returns the desired password. I think it can avoid ```segmentation fault```. Changes coming soon.

Reply:
The order of definition does not matter here as the destructor is invoked automatically when the relative object goes out of scope. I meant it would make sense if it is defined at the end as well.